tensorflow/core/kernels/relu_op_gpu.cu.cc

```diff
@@ -119,12 +119,22 @@ struct ReluGrad<Device, Eigen::half> {
 };
 #endif  // GOOGLE_CUDA
 
-#if GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 __global__ void Relu_int8x4_kernel(int vect_count,
                                    const int32* __restrict__ input,
                                    int32* __restrict__ output) {
   CUDA_1D_KERNEL_LOOP(index, vect_count) {
+#if GOOGLE_CUDA    
     output[index] = __vmaxs4(input[index], 0);
+#else
+    uint32 signs = (~input[index]) & 0x80808080;
+    signs = signs>>7;
+    signs |= signs<<1;
+    signs |= signs<<2;
+    signs |= signs<<4;
+    signs &= 0x7f7f7f7f;
+    output[index] = input[index] & signs;
```

Comment:
very clever :)

thanks for looking into this

Reply:
/cc @whchung 

Jack, Who is the right person to talk to w.r.t. the status of int8x4 support in ROCm?

wondering whether it makes sense to add this change as `__vmaxs4` routine in the `hip/hcc_details/math_functions.h` file and then call it here?

Reply:
nevermind....the change above is not really "max" functionality, so will not work as a generic `__vmaxs4` routine.